### ISSUE TITLE
Contact stops updating for some reason

### ISSUE THREAD

**Issue by KevinDenys** [JohanDegraeve/xdripswift#537]
For the second time my contact stops updating somewhere in the middle of the night. The only thing I can do to fix it is to re-enable the contact feature. I exported the logs and saw the last log of the contactImageManager was at 1h10

![image](https://github.com/JohanDegraeve/xdripswift/assets/17174443/d7567246-d185-4406-bf20-c8570efe2b65)

And after that no more contactImageManager logs. When I woke up in the morning I saw the contact image was not in sync with the current value

**Comment by KevinDenys** [comment]
[xdriptrace.0.log](https://github.com/JohanDegraeve/xdripswift/files/15259591/xdriptrace.0.log)

**Comment by paulplant** [comment]
Had a quick look, this will help.

Is there anything else you can tell me? Were you actively using your phone around (or just before) it stopped updating at 01:10hrs? No changes to the heartbeat settings or anything?

Please keep monitoring whilst I get time to check things.

**Comment by KevinDenys** [comment]
Nope, we went to sleep around 22.30h. Ill keep monitoring it when it stops syncing

**Comment by KevinDenys** [comment]
Stopped again 

i guess something with bad connection?
![IMG_5418](https://github.com/JohanDegraeve/xdripswift/assets/17174443/b725982d-921f-4b06-8726-0eb9ce278da0)

![IMG_5419](https://github.com/JohanDegraeve/xdripswift/assets/17174443/f5c16125-0324-4fe9-bd23-ca2023cbff40)

**Comment by KevinDenys** [comment]
I also can reproduce it with just hard closing the xdrip app. It does not restart the contact sync after reopening the xdrip app

**Comment by paulplant** [comment]
> I also can reproduce it with just hard closing the xdrip app. It does not restart the contact sync after reopening the xdrip app

Interesting. Thanks!

**Comment by dugamarian** [comment]
Hello, I'm encountering the same issue. I've restarted my phone, rebuilt in Xcode, but it still doesn't automatically update the contact. Do you have any suggestions?
![IMG_6399](https://github.com/JohanDegraeve/xdripswift/assets/35396265/31c62fa0-93f6-4e15-aa20-5b409b33a3fd)

**Comment by paulplant** [comment]
> Hello, I'm encountering the same issue. I've restarted my phone, rebuilt in Xcode, but it still doesn't automatically update the contact. Do you have any suggestions?

I'm not sure it's the same issue?

Your contact was updated just 8 minutes earlier than the time in screenshot? And you had enabled the Contact Image function (and created the contact) just 5 minutes before that.

So what exactly isn't working? The screenshot you've shown would indicate that it was working until 3 minutes earlier?

**Comment by dugamarian** [comment]
Yes, I've tried a bunch of things hoping it might work.
It seems to be working after all, but it updates every few minutes rather than minute by minute.
it’s a normal behavior?
![IMG_6401](https://github.com/JohanDegraeve/xdripswift/assets/35396265/68d5a008-e7e6-4db3-8c8c-60363d5b8329)
![IMG_6400](https://github.com/JohanDegraeve/xdripswift/assets/35396265/ef30bcbb-b587-4bd5-8dd8-7eb49c424b1e)

**Comment by paulplant** [comment]
I've never tested it with 60 second updates to be honest.

But again, this issue is not related to the original issue post that was opened where xDrip4iOS wasn't updating the contact image at all...

**Comment by dugamarian** [comment]
Okay. But how often does it update?

**Comment by nahog** [comment]
This is happening fairly consistent to me. To the case where I usually don’t rely anymore on the contact image. I’m not sure if my use case can provide some light to the issue. But I have an Android phone that uploads to Nightscout (using CamAPS+xDrip notifications) but the android phone does not have 100% connection as it’s only connected to wifi. So Shuggah (on my iPhone with mobile data) loses chunks of updates until my android phone connects to wifi. These timeframes without data I think makes the contact image to stops updating until y check/uncheck the “create contact” toggle.

**Comment by paulplant** [comment]
@KevinDenys , does this issue need to be kept open still?

There is a WatchOS issue where the contact complication can (for no apparent reason), simply stop updating after some time (some time = 2 days or 2 weeks).

Once the Watch is restarted, it is fixed until it happens again.

What I mean is that this issue is often attributed to being a problem with the apps using the Contact Image even though it is unrelated at all.

**Comment by paulplant** [comment]
I'll close as I'm guessing it's no longer needed. Can be reopened as required...

**Comment by timmy354321** [comment]
I’d like to join back into this as this happened to me last night. The contact image has been updating consistently then at around 1 am it stopped updating. When I went into the contacts app I see that there are 2 xdrip contacts. The first one (on my watch and not updating) and a second one that is/was updating. Had to delete the old contact and selected the new one on my watch.
